DialogHost.Avalonia/DialogHost.axaml.cs

```diff
@@ -630,34 +655,130 @@ protected void RaiseCommandsCanExecuteChanged() {
         }
 
         _templateDisposables = new CompositeDisposable() {
             // this.GetObservable(BoundsProperty)
             // .Subscribe(rect => _overlayPopupHost?.ConfigurePosition(_root, PlacementMode.AnchorAndGravity, new Point())),
-            _overlayPopupHost!.Bind(DisableOpeningAnimationProperty, this.GetBindingObservable(DisableOpeningAnimationProperty)),
-            _overlayPopupHost!.Bind(ContentProperty, this.GetBindingObservable(DialogContentProperty)),
-            _overlayPopupHost!.Bind(ContentTemplateProperty, this.GetBindingObservable(DialogContentTemplateProperty)),
-            _overlayPopupHost!.Bind(TemplateProperty, this.GetBindingObservable(PopupTemplateProperty)),
-            _overlayPopupHost!.Bind(PaddingProperty, this.GetBindingObservable(DialogMarginProperty)),
-            _overlayPopupHost!.Bind(PopupPositionerProperty, this.GetBindingObservable(PopupPositionerProperty)),
+            _disposeList,
             e.NameScope.Find<Rectangle>(ContentCoverName)?.AddDisposableHandler(PointerReleasedEvent, ContentCoverGrid_OnPointerReleased) ?? EmptyDisposable.Instance
         };
         base.OnApplyTemplate(e);
     }
 
+    private class DisposeList : IDisposable {
+        private readonly Dictionary<DialogOverlayPopupHost, List<IDisposable>> _hostDisposeList = [];
+
+        public void Dispose() {
+            foreach (var list in _hostDisposeList.Values) {
+                foreach (var item in list) {
+                    item.Dispose();
+                }
+            }
+            _hostDisposeList.Clear();
+        }
+
+        public void AddDispose(DialogOverlayPopupHost host, params IDisposable[] disposables) {
+            _hostDisposeList[host] = [.. disposables];
+        }
+
+        public void RemoveDispose(DialogOverlayPopupHost host) {
+            if (_hostDisposeList.TryGetValue(host, out var list)) {
+                _hostDisposeList.Remove(host);
+                foreach (var item in list) {
+                    item.Dispose();
+                }
+            }
+        }
+    }
+
+    private TaskCompletionSource<object?> AddHost(object? content) {
+        foreach (var item in _overlayPopupHosts) {
+            if (item.Content == content) {
+                return item.DialogTaskCompletionSource;
+            }
+        }
+
+        var host = new DialogOverlayPopupHost(this) {
+            Content = content, ContentTemplate = DialogContentTemplate, Template = PopupTemplate,
+            Padding = DialogMargin, ClipToBounds = false, DisableOpeningAnimation = DisableOpeningAnimation,
+            PopupPositioner = PopupPositioner
+        };
+
+        _disposeList.AddDispose(host, 
+            host.Bind(DisableOpeningAnimationProperty, this.GetBindingObservable(DisableOpeningAnimationProperty)),
+            //host.Bind(ContentProperty, this.GetBindingObservable(DialogContentProperty)),
```

Comment:
Probably we can bind Content property if no content passed from user. It allows us to decouple DialogContent from user called dialogs. I can do it myself, actually